src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
             },
             "file_name_col": {
                 "type": "string",
-                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date."
+                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date.",
             },
             "map": {
                 "type": "list",
                 "doc": "List of lists of unique publications references and publication names in the collection provided in the same order.",
                 "optional": True,
-            }
+            },
         }
 
     def create_outputs_schema(self):
         return {
             "corpus_table": {
                 "type": "table",
-                "doc": "Augmented table containing extracted metadata."
+                "doc": "Augmented table containing extracted metadata.",
             }
         }
 
     def process(self, inputs, outputs) -> None:
-        
+        import re
+        import polars as pl
+        import pyarrow as pa
+
         table_obj = inputs.get_value_obj("corpus_table")
         column_name = inputs.get_value_obj("file_name_col").data
-        pub_refs, pub_names = None, None
 
-        try:
-            pub_refs = inputs.get_value_obj("map").data[0]
-            pub_names = inputs.get_value_obj("map").data[1]
+        if not column_name:
+            raise KiaraProcessingException("No file name column name provided.")
+        
+        if not table_obj:
+            raise KiaraProcessingException("No table provided.")
+        
+        if table_obj.is_set:
 
-        except Exception as e:
-            pass;
+            sources: Union[pa.Table, None] = table_obj.data
```

Comment:
So, following on from above, we now know that the `.data` property can't be None (since 'optional' wasn't set), so in my mind it would be ok to do:

```
sources: KiaraTable = table_obj.data  # type: ignore
```
The `type: ignore` is justified since we can reasonable assume kiara has tested for `None` before we even entered the `process` method.
The other thing is that the kiara `table` data type has a `KiaraTable` Python data type, not `pa.Table`, That's a wrapper where I copied some of the properties of the arrow table type for easier access, which is why the type checker doesn't complain, since `column_names` exists in both. Otherwise it would have picked it up I think.

Reply:
Would I then be able to do sources.column_names to check the columns?
This is why I only retrieved the ".data" afterwards.

Reply:
> The other thing is that the kiara table data type has a KiaraTable Python data type, not pa.Table

Ok, I thought that the Kiara data table could contain several types of tabular data and that it would be good to specify which one is expected here (since the module assumes an Arrow table), but it is not the case if I understand right?

Reply:
> Would I then be able to do sources.column_names to check the columns? This is why I only retrieved the ".data" afterwards.

Yes, you just have to check what methods the [`KiaraTable`](https://github.com/DHARPA-Project/kiara_plugin.tabular/blob/178fcb6914fdaf0f10dfed92492a69dbd593a985/src/kiara_plugin/tabular/models/table.py#L30) class offers. [`column_names`](https://github.com/DHARPA-Project/kiara_plugin.tabular/blob/178fcb6914fdaf0f10dfed92492a69dbd593a985/src/kiara_plugin/tabular/models/table.py#L106) is available, so... Usually, if you have a type hint, your IDE would let you know what you can use, incl. docs of the property/method. Mine does, anyway. So no need to check out the source code manually. Sometimes I 'jump to definition' to read the source of what I'm calling to understand it better, that's also something I'd recommend figuring out how to do it in your IDE of choice, it's very powerful.

Reply:
> Ok, I thought that the Kiara data table could contain several types of tabular data and that it would be good to specify which one is expected here (since the module assumes an Arrow table), but it is not the case if I understand right? I then don't really see the added value of the hint in this case, since the table type is already specified in input documentation.

The `KiaraTable` class (like many of the kiara implementation classes for data types) just wraps the actual implementation of whatever Python type was chosen to 'represent' the *kiara* data type (and we could change that under the hood without the user ever knowing). This is an additional level of abstraction that I found beneficial because I can augment the data type with convenience methods, and have better control over which 'characteristics' of data I want to control in what way. And I can improve the data type over time to new kiara-specific requirements. In theory, I could have just used a Pandas DataFrame class to represent the *kiara* table type, but in that case I'd have no control over the implementation, and would have to hope that Pandas doesn't do anything incompatible in the future.

 If you look up the `KiaraTable` class, you'll see that it has a few convenience methods, to export to a pandas data frame, or polars, or arrow. That's when you'd annotate the Python type, like:
 
```
kiara_table_instance: KiaraTable = inputs.get_value_data('table') # this also has `Any` annotated as result, same as the `value.data` attribute

arrow_table: pa.Table = kiara_table_instance.arrow_table
# or
pandas_table: pd.DataFrame = kiara_table_instance.to_pandas_dataframe()
# or
polars_table: pl.DataFrame = kiara_table_instance.to_polars_dataframe()
# or even
python_dict: Dict = kiara_table_instance.to_pydict()
```

Sidenote: I chose to use method names with `to_` in the name to indicate that there is a conversion happening, that could potentially be expensive (it's usually not for polars, but might be for pandas).

Technically, as I've written above, you don't really need to annotate here, because the type checker can infer the Python type from the method call results. But I consider it good practice to at least name the variable to contain which type it is, keeps things clear I find. But I wouldn't die on that hill, so use your own judgement there.